--- a/Source/Compiler/NarayanLogicLexerBase.cpp
+++ b/Source/Compiler/NarayanLogicLexerBase.cpp
@@ -8,6 +8,9 @@
 // $insert class_ih
 #include "NarayanLogicLexer.ih"
 
+// Added manually. Declares all tokens both the lexer and parser use...
+#include "Compiler/NarayanLogicParserBase.h"
+
 
     // s_ranges__: use (unsigned) characters as index to obtain
     //           that character's range-number.
