--- a/Source/Compiler/NarayanLogicParser.ih
+++ b/Source/Compiler/NarayanLogicParser.ih
@@ -22,7 +22,7 @@ inline void NarayanLogicParser::print()
     print__();           // displays tokens if --print was specified
 }
 
-inline void NarayanLogicParser::exceptionHandler(std::exception const &exc)         
+inline void NarayanLogicParser::exceptionHandler(std::exception const &/*exc*/)
 {
     throw;              // re-implement to handle exceptions thrown by actions
 }
